docs/source/simulation.dox

```diff
@@ -0,0 +1,100 @@
+namespace smaug {
+/**
+  ***********************************************************************
+  ***********************************************************************
+
+\page simulation SMAUG in simulation
+
+Due to certain limitations of the syscall-emulation mode of gem5-Aladdin, there
+are two key differences in how SMAUG runs under simulation compared to on real
+hardware, which affect how you will write code for SMAUG.
+
+\section sampling Sampling of accelerated kernels
+
+DL models are highly compute-intensive, but oftentimes these core kernels are
+very repetitive, so we can afford to sample the loop iterations to save on
+simulation time and trace storage. This means that during tracing and
+simulation, we will only execute a small fraction of the total loop iterations,
+and afterwards, we unsample this simulated time to estimate the actual time
+spent in the kernel. If sampling is enabled, functional results will be
+incorrect. This has implications if sampled operators feed into operators whose
+control-flow is data-dependent, so please keep this in mind.
```

Comment:
Maybe add a number from the SMAUG paper that sampling for DNNs actually introduce small performance errors.

Reply:
Done.